Further changes to the RTC-S1 SAS after the Cal/Val Release

gshiroma opened the pull request:
This PR updates the RTC-S1 SAS:
- Separate RTC-S1 and RTC-S1 static layers (SL) runconfigs;
- Add new layer: "RTC area normalization factor (ANF) gamma0 to sigma0" to enable users to normalize RTC-S1 backscatter from gamma0 to sigma0;
- Use constants to represent product layers;
- Fix bugs;
- Lint code.

seongsujeong commented on src/rtc/rtc_s1_single_job.py:
```diff
@@ -19,77 +19,34 @@
 from rtc.h5_prep import (save_hdf5_file, create_hdf5_file,
                          get_metadata_dict,
                          all_metadata_dict_to_geotiff_metadata_dict,
+                         layer_hdf5_dict,
```
Looks like `layer_hdf5_dict` was not used in this code. Can you check?

seongsujeong commented on src/rtc/schemas/rtc_s1.yaml:
```diff
@@ -2,6 +2,10 @@ runconfig:
 
     groups:
+
+        primary_executable:
```
I think it would be more informative to describe what `primary_executable` is.

seongsujeong commented on src/rtc/runconfig.py:
```diff
@@ -139,19 +133,30 @@ def load_validate_yaml(yaml_path: str, workflow_name: str) -> dict:
+        'product_type']
+    if (product_type == STATIC_LAYERS_PRODUCT_TYPE):
+        default_cfg_path = (f'{helpers.WORKFLOW_SCRIPTS_DIR}/defaults/'
+                            f'rtc_s1_sl.yaml')
```
What is your decision about the workflow name for the static layer processing after the ADT/SDS meeting last Friday?

gshiroma replied:
Thank you, @seongsujeong . It will probably become `RTC_S1_STATIC` rather than `RTC_S1_SL`, but if possible, I'd like to handle this in another PR after I talk to Heresh. In the next PR, I also intend to change the way we define the radar grid for static layers (so that the processing will not be 12x slower).

seongsujeong reviewed (commented):
Here is my initial comments and suggestions for the PR. Most of them are minor. - Seongsu

gshiroma commented on src/rtc/h5_prep.py:
```diff
@@ -27,6 +27,106 @@
 
 logger = logging.getLogger('rtc_s1')
 
+LAYER_NAME_VV = 'VV'
+LAYER_NAME_VH = 'VH'
+LAYER_NAME_LAYOVER_SHADOW_MASK = 'layover_shadow_mask'
+LAYER_NAME_RTC_ANF_GAMMA0_TO_BETA0 = 'rtc_anf_gamma0_to_beta0'
+LAYER_NAME_RTC_ANF_GAMMA0_TO_SIGMA0 = 'rtc_anf_gamma0_to_sigma0'
+LAYER_NAME_RTC_ANF_SIGMA0_TO_BETA0 = 'rtc_anf_sigma0_to_beta0'
+LAYER_NAME_RTC_ANF_BETA0_TO_BETA0 = 'rtc_anf_beta0_to_beta0'
+LAYER_NAME_NUMBER_OF_LOOKS = 'number_of_looks'
+LAYER_NAME_INCIDENCE_ANGLE = 'incidence_angle'
+LAYER_NAME_LOCAL_INCIDENCE_ANGLE = 'local_incidence_angle'
+LAYER_NAME_PROJECTION_ANGLE = 'projection_angle'
+LAYER_NAME_RTC_ANF_PSI = 'rtc_anf_psi'
+LAYER_NAME_RANGE_SLOPE = 'range_slope'
+LAYER_NAME_DEM = 'dem'
```
Thank you, @seongsujeong . What would be the keys for that dictionary? You need to define them somewhere. These constants define the keys that are used in the dictionaries defined below.

seongsujeong replied:
I see. So it look like those constants are used as keys in the following dictionaries (e.g. `layer_hdf5_dict`, `layer_names_dict`), as well as layer names in `rtc_single_jb.py` and etc.
That being case, I think the dictionary name is `LAYER_NAME_DICT`, and keys are `VV`, `VH`, `LAYER_SHADOW_MASK` and so on.

The point that I am suggesting here is to bundle the layer name constants to organize them better.

If you prefer, I think `SimpleNamespace` instead of `dictionary` would be okay to bundle the layer names here.

seongsujeong replied:
Sounds good. Let's work on that after merging this PR.

gshiroma replied:
Thank you, @seongsujeong . Please. let me know if you have any other comments or concerns.

gshiroma commented:
> Here is my initial comments and suggestions for the PR. Most of them are minor. - Seongsu

Thank you, @seongsujeong . Just addressed your comments. Please, let me know if you have any other comments or concerns.

seongsujeong commented on src/rtc/h5_prep.py:
```diff
@@ -27,6 +27,112 @@
+                    ' coefficient normalized to gamma0'),
+    LAYER_NAME_VH: ('Radiometric terrain corrected Sentinel-1 VH backscatter'
+                    ' coefficient normalized to gamma0'),
+    LAYER_NAME_LAYOVER_SHADOW_MASK: ('Layover/shadow mask. Values: 0: not'
```
Maybe you'll need to add the description that `255` is invalid NODATA value?

seongsujeong reviewed (approved):
Just a few minor suggestions. Other than that the PR looks good to me, so I approve this.
Please take care of the issues that I have pointed out when testing this PR. - Seongsu
